src/agent/debuglet.ts

```diff
@@ -405,6 +406,38 @@ export class Debuglet extends EventEmitter {
         });
   }
 
+  static async getProjectId(config: DebugAgentConfig): Promise<string> {
+    const project = config.projectId ||
+        process.env.GCLOUD_PROJECT ||
+        await this.getProjectIdFromMetadata();
+    if (!project) {
+      const msg = 'Unable to discover projectId. Please provide the ' +
+          'projectId to be able to use the Debug agent';
+      throw new Error(msg);
+    }
+    return project;
+  }
+
+  static async runningOnGCP(): Promise<boolean> {
+    const lookup = promisify(dns.lookup);
+    try {
+      await lookup('metadata.google.internal.');
+      return true;
+    } catch (err) {
```

Comment:
We should probably have a comment here describing why it is ok to ignore `err` so that, looking back, we know it was intentional.